packages/decoder/lib/Web3Adapter.ts

```diff
@@ -0,0 +1,50 @@
+import Web3Eth from "web3-eth";
+type PastLogsOptions = {
```

Comment:
I lifted most of this type from the web3 types, I don't know if we can get rid of the optional arguments as I don't know whether the decoder always includes these or not.

Reply:
Worth noting, the web3 types are kind of unreliable atm; that's why we use `@types/web3` rather than trusting the types it comes with.  (When I tried switching to that, it went badly.)  They've been planning to fix this but I don't think they've done so yet?